aten/src/ATen/function_wrapper.py

```diff
@@ -760,6 +769,26 @@ def gen_dispatch_key_init(var_name, formals):
     ]
 
 
+def is_factory(option):
+    # type: (FunctionOption) -> bool
+    formals = option['formals_list']
+    return find_formal_by_type('TensorOptions', formals) is not None and 'method' not in option['variants']
```

Comment:
I guess these are all non-substantive refactors?

Reply:
Yeah - sorry about the noisy diff. This just factors out the old RHS of `is_factory_method` (line 1224) so it can be used to gate the device init code as well. `find_formal_by_name` is just the old `find_formal` moved up and out to accompany.